exporter/terra/spreadsheet/exporter.py

```diff
@@ -22,24 +23,30 @@ def __init__(self, ingest_service: IngestService, terra_client:  TerraStorageCli
         self.logger = logging.getLogger('IngestSpreadsheetExporter')
 
     def export_spreadsheet(self, job_id: str, project_uuid: str, submission_uuid: str):
-        self.logger.info("Generating Spreadsheet")
-        workbook = self.downloader.get_workbook_from_submission(submission_uuid)
-        self.logger.info("Generating Metadata")
-        project = self.ingest.get_metadata(entity_type='projects', uuid=project_uuid)
-        with NamedTemporaryFile() as spreadsheet:
-            workbook.save(spreadsheet.name)
-            file = self.create_supplementary_file_metadata(spreadsheet, project)
-            self.logger.info("Writing to Terra")
-            self.write_to_terra(spreadsheet, project, file)
+        try:
+            self.logger.info("Generating Spreadsheet")
+            workbook = self.downloader.get_workbook_from_submission(submission_uuid)
+            self.logger.info("Generating Spreadsheet Metadata")
+            project = self.ingest.get_metadata(entity_type='projects', uuid=project_uuid)
+            with NamedTemporaryFile() as spreadsheet_file:
+                workbook.save(spreadsheet_file.name)
+                # todo: make it available in broker as well.
+                file = self.create_supplementary_file_metadata(spreadsheet_file, project)
+                self.logger.info("Writing to Terra")
+                self.write_to_terra(spreadsheet_file, project, file)
+        except Exception as e:
+            self.logger.exception(f'problem generating spreadsheet for {project_uuid}')
+            raise SpreadsheetExportError from e
```

Comment:
putting this try-catch here doesn't add anything since the error will already be logged **with the correct context** by
[The Queue Listener](https://github.com/ebi-ait/ingest-exporter/blob/0947b61d319912a21e865bf9dcefe5a6291506ec/exporter/queue/listener.py#L34)